source/Session.cpp

```diff
@@ -121,6 +121,29 @@ bool Session::hasNextCondition() const{
 	return false;
 }
 
+const RealTime Session::targetFrameTime()
+{
+	if (m_config->render.frameTimeArray.size() == 0) {
+		// The below matches the functionality in FPSciApp::updateParameters()
+		if (m_config->render.frameRate > 0) {
+			return 1.0f / m_config->render.frameRate;
+		}
+		else {
+			return 1.0f / float(m_app->window()->settings().refreshRate);
+		}
+	}
+	else {
+		if (m_config->render.randomFrameTime) {
+			return m_config->render.frameTimeArray.randomElement();
+		}
+		else {
+			static uint targetIdx = 0;
+			return m_config->render.frameTimeArray[targetIdx++ % m_config->render.frameTimeArray.size()];
+		}
+	}
+	return 0.0;
```

Comment:
Seems like this would never be hit, I suppose this value would indicate "don't wait at all" correct?

Reply:
I agree it should never be hit, but there's probably a better fallback. I'll revise.

Reply:
This is likely just fine. I guess the `else` case from the `if` above (`1.0f / float(m_app->window()->settings().refreshRate)`) could be this default...